src/server/game/Entities/AreaTrigger/AreaTrigger.cpp

```diff
@@ -756,8 +756,14 @@ void AreaTrigger::DoActions(Unit* unit)
                     case AREATRIGGER_ACTION_TELEPORT:
                     {
                         if (WorldSafeLocsEntry const* safeLoc = sObjectMgr->GetWorldSafeLoc(action.Param))
+                        {
                             if (Player* player = caster->ToPlayer())
+                            {
+                                if (WorldSafeLocsEntry const* instanceEntrance = player->GetInstanceEntrance(safeLoc->Loc.GetMapId()))
```

Comment:
So this makes having teleporting areatriggers impossible to use inside instances with all of them pointing to the entrance location now?

Reply:
fixed